Source/WebCore/page/EventHandler.cpp

```diff
@@ -2346,6 +2346,11 @@ HandleUserInputEventResult EventHandler::handleMouseReleaseEvent(const PlatformM
         return true;
     }
 
+    if (m_lastElementUnderMouse && &m_lastElementUnderMouse->document() != frame->document()) {
+        m_lastElementUnderMouse = nullptr;
+        m_lastScrollbarUnderMouse = nullptr;
+    }
```

Comment:
We could have cleared out `m_lastElementUnderMouse` at the time the element moved, rather than waiting until `EventHandler::handleMouseReleaseEvent` is called. There is a `Document` function called when an element is removed from a document, and this might be efficient enough to call there.

Reply:
Thanks very much for the suggestion! Code updated. PTAL.